master/internal/db/postgres_experiments.go

```diff
@@ -194,28 +195,31 @@ func snapshotWrapperToTrial(r snapshotWrapper) (*apiv1.TrialsSnapshotResponse_Tr
 	}
 	trial.Hparams = protoutils.ToStruct(inter)
 	trial.Metric = r.Metric
+	trial.BatchesProcessed = r.Batches
 	return &trial, nil
 }
 
 // TrainingTrialsSnapshot returns a training metric across each trial in an experiment at a
 // specific point of progress.
-func (db *PgDB) TrainingTrialsSnapshot(experimentID int, batchesProcessed int, metricName string,
-	startTime time.Time) (trials []*apiv1.TrialsSnapshotResponse_Trial, endTime time.Time,
-	err error) {
+func (db *PgDB) TrainingTrialsSnapshot(experimentID int, minBatches int, maxBatches int,
+	metricName string, startTime time.Time) (trials []*apiv1.TrialsSnapshotResponse_Trial,
+	endTime time.Time, err error) {
 	var rows []snapshotWrapper
 	err = db.queryRows(`
 SELECT
   t.id AS trial_id,
   t.hparams AS hparams,
   s.metrics->'avg_metrics'->$1 AS metric,
-  s.end_time AS end_time
+  s.end_time AS end_time,
```

Comment:
bit of a nit, but now these query can potentially return trial id and hparams multiple times per trial. could get a little inventive with an agg to prevent that but i think then it would logically follow to change the API too, which we probably don't want to do. how large of ranges are we planning on selecting? e.g. if a user runs a 3 day adaptive with a lot of hparams then calls this for min=0 max=inf are they going to oom the master?

Reply:
Yeah this has been discussed. If they use the API (which is marked as "Internal", fwiw) they could select a massive range and just return all metrics. On the front-end we're going to have a drop-down to select a few pre-defined ranges OR a heuristic that selects the range for us.

The reason for adding this is specifically to deal with fairly small ranges: you want to select a point of progress in the trial, and we'll show all trials that recorded data at that particular point. But little cases pop up where there's only 1 trial at each point, but they're just a couple of batches offset from each other.

Reply:
> On the front-end we're going to have a drop-down to select a few pre-defined ranges OR a heuristic that selects the range for us.

any way to enforce these ranges are well behaved master side?

Reply:
Still wanting to confirm with AMLEs, but we discussed a reasonable limit that's above what the front-end will suggest.